diff --git a/Modules/_multiprocessing/semaphore.c b/Modules/_multiprocessing/semaphore.c
index 8c6a4a27920..ae1ecc9e551 100644
--- a/Modules/_multiprocessing/semaphore.c
+++ b/Modules/_multiprocessing/semaphore.c
@@ -631,14 +631,14 @@ _multiprocessing_SemLock__get_value_impl(SemLockObject *self)
     PyErr_SetNone(PyExc_NotImplementedError);
     return NULL;
 #else
-    int sval;
+    long sval;
     if (SEM_GETVALUE(self->handle, &sval) < 0)
         return _PyMp_SetError(NULL, MP_STANDARD_ERROR);
     /* some posix implementations use negative numbers to indicate
        the number of waiting threads */
     if (sval < 0)
         sval = 0;
-    return PyLong_FromLong((long)sval);
+    return PyLong_FromLong(sval);
 #endif
 }
 
@@ -663,10 +663,10 @@ _multiprocessing_SemLock__is_zero_impl(SemLockObject *self)
         Py_RETURN_FALSE;
     }
 #else
-    int sval;
+    long sval;
     if (SEM_GETVALUE(self->handle, &sval) < 0)
         return _PyMp_SetError(NULL, MP_STANDARD_ERROR);
-    return PyBool_FromLong((long)sval == 0);
+    return PyBool_FromLong(sval == 0);
 #endif
 }
 
diff --git a/Modules/posixmodule.c b/Modules/posixmodule.c
index d81f59df5c0..eea211b3b8c 100644
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -5163,7 +5163,7 @@ os__path_splitroot_impl(PyObject *module, path_t *path)
 /*[clinic end generated code: output=ab7f1a88b654581c input=dc93b1d3984cffb6]*/
 {
     wchar_t *buffer;
-    wchar_t *end;
+    const wchar_t *end;
     PyObject *result = NULL;
     HRESULT ret;
 
@@ -6966,9 +6966,9 @@ os_execv_impl(PyObject *module, path_t *path, PyObject *argv)
 
     _Py_BEGIN_SUPPRESS_IPH
 #ifdef HAVE_WEXECV
-    _wexecv(path->wide, argvlist);
+    _wexecv(path->wide, (const EXECV_CHAR *const *)argvlist);
 #else
-    execv(path->narrow, argvlist);
+    execv(path->narrow, (const EXECV_CHAR *const *)argvlist);
 #endif
     _Py_END_SUPPRESS_IPH
 
@@ -7054,9 +7054,9 @@ os_execve_impl(PyObject *module, path_t *path, PyObject *argv, PyObject *env)
     else
 #endif
 #ifdef HAVE_WEXECV
-        _wexecve(path->wide, argvlist, envlist);
+        _wexecve(path->wide, (const EXECV_CHAR *const *)argvlist, (const EXECV_CHAR *const *)envlist);
 #else
-        execve(path->narrow, argvlist, envlist);
+        execve(path->narrow, (const EXECV_CHAR *const *)argvlist, (const EXECV_CHAR *const *)envlist);
 #endif
     _Py_END_SUPPRESS_IPH
 
@@ -7706,11 +7706,11 @@ os_spawnv_impl(PyObject *module, int mode, path_t *path, PyObject *argv)
     Py_BEGIN_ALLOW_THREADS
     _Py_BEGIN_SUPPRESS_IPH
 #ifdef HAVE_WSPAWNV
-    spawnval = _wspawnv(mode, path->wide, argvlist);
+    spawnval = _wspawnv(mode, path->wide, (const EXECV_CHAR *const *)argvlist);
 #elif defined(HAVE_RTPSPAWN)
     spawnval = _rtp_spawn(mode, path->narrow, (const char **)argvlist, NULL);
 #else
-    spawnval = _spawnv(mode, path->narrow, argvlist);
+    spawnval = _spawnv(mode, path->narrow, (const EXECV_CHAR *const *)argvlist);
 #endif
     _Py_END_SUPPRESS_IPH
     Py_END_ALLOW_THREADS
@@ -7822,12 +7822,12 @@ os_spawnve_impl(PyObject *module, int mode, path_t *path, PyObject *argv,
     Py_BEGIN_ALLOW_THREADS
     _Py_BEGIN_SUPPRESS_IPH
 #ifdef HAVE_WSPAWNV
-    spawnval = _wspawnve(mode, path->wide, argvlist, envlist);
+    spawnval = _wspawnve(mode, path->wide, (const EXECV_CHAR *const *)argvlist, (const EXECV_CHAR *const *)envlist);
 #elif defined(HAVE_RTPSPAWN)
     spawnval = _rtp_spawn(mode, path->narrow, (const char **)argvlist,
                            (const char **)envlist);
 #else
-    spawnval = _spawnve(mode, path->narrow, argvlist, envlist);
+    spawnval = _spawnve(mode, path->narrow, (const EXECV_CHAR *const *)argvlist, (const EXECV_CHAR *const *)envlist);
 #endif
     _Py_END_SUPPRESS_IPH
     Py_END_ALLOW_THREADS
diff --git a/PC/launcher.c b/PC/launcher.c
index ffd7f7ee8f2..23072e0e4d7 100644
--- a/PC/launcher.c
+++ b/PC/launcher.c
@@ -1914,7 +1914,7 @@ process(int argc, wchar_t ** argv)
     {
         FILE *f;
         char buffer[4096]; /* 4KB should be enough for anybody */
-        char *start;
+        const char *start;
         DWORD len, cch, cch_actual;
         size_t cb;
         if (_wfopen_s(&f, venv_cfg_path, L"r")) {
